### PR TITLE
Add phpcsfixer

### PR DESCRIPTION
**Pull request by TomKlotzPro**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no    
| BC breaks?        | no     
| Related Issue     | https://github.com/algolia/algoliasearch-client-php/issues/519
| Need Doc update   | no

## Describe your change
This proposal adresses about **implementing a quality tool to check syntax.**
> This proposal was already discussed in https://github.com/algolia/algoliasearch-client-php/issues/519

Currently we're using [StyleCI](https://styleci.io/) to check syntax after we've committed our modification. 

Implementing [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/tree/v2.0.0) would be the best quality tool to implement and improve our delivery because we can see all conflicts before a commit.

I've set the same configuration as [StyleCI](https://styleci.io/). 

**Here is the result**
Screenshot 2019-03-14 at 11 27 42
_See that it show warning on files that are passing for [StyleCI](https://styleci.io/)._

**Here is the result**
Screenshot 2019-03-14 at 11 33 58

There are *many rules* that can be set but in fact by pre-setting `@Symphony`we handle many of them.
> Perhaps open a discussion on what we want to add, i think this configuration is enough for us. 

I want to implement the version [2.0.0](https://github.com/FriendsOfPHP/PHP-CS-Fixer/tree/v2.0.0), it support php 5.3 and all versions *above*.

## What problem is this fixing?

* Having too much commit 
* Improve the quality of our code 
* Add better quality tool

### CODE REVIEW

**Comment by TomKlotzPro** [comment]
Need a review @nunomaduro

File: composer.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
         "psr/simple-cache": "^1.0"
     },
     "require-dev": {
+        "friendsofphp/php-cs-fixer": "^2.0",
```
Comment (nunomaduro) [inline composer.json]: The pull request shouldn't contain a pre-defined set of rules? At StyleCI we have adopted the Symfony ones.

Comment (nunomaduro) [reply composer.json]: You should also define a job on `.travis.yml` here the command will have a `exit code > 0`, if the coding style is not correct.

Comment (nunomaduro) [reply composer.json]: Don't forget that the output should be a table with the errors.

Comment (TomKlotzPro) [reply composer.json]: @nunomaduro the pre-defined set of rules was already made like StyleCI, i've modified it and conclude that the original one was really good but perhaps open the discussion on adding rules 

Ok ! i'll find out how to manage to have a table

**Comment by TomKlotzPro** [comment]
need a review 👍  @nunomaduro

**Comment by TomKlotzPro** [comment]
Travis is failing.
> $ ./vendor/bin/php-cs-fixer fix -v --dry-run
/home/travis/.travis/functions: line 104: ./vendor/bin/php-cs-fixer: No such file or directory

perhaps it's because i should write `vendor/bin/php-cs-fixer fix -v --dry-run`

@nunomaduro

**Comment by nunomaduro** [comment]
@TomKlotzPro Maybe! give it a try.

**Comment by TomKlotzPro** [comment]
Travis : `Parse error: syntax error, unexpected '[' in /home/travis/build/algolia/algoliasearch-client-php/.php_cs.dist on line 12 `

I'm changing the bracet by array();

**Comment by TomKlotzPro** [comment]
This rule ` 'increment_style' => array('style' => 'post')` does not exist in version [2.0] (https://github.com/FriendsOfPHP/PHP-CS-Fixer/tree/v2.0.0) 

I removing it.

**Comment by TomKlotzPro** [comment]
Php-cs-fixer report error **phpdoc_types_order** from two files thats are not correct. 
* `src/Http/Psr7/Response.php `
* `src/Http/Psr7/Request.php`

Perhaps open an issues to correct them.

**Comment by nunomaduro** [comment]
I think you can correct those issues in this pull request.

**Comment by TomKlotzPro** [comment]
Last commit was bad, sorry. 

I've Corrected header in Request and Responses files

File: .php_cs.dist [hunk 2]
```diff
@@ -4,13 +4,28 @@ $finder = \PhpCsFixer\Finder::create()
     ->in(__DIR__ . DIRECTORY_SEPARATOR . 'src')
     ->in(__DIR__ . DIRECTORY_SEPARATOR . 'tests')
```
Comment (nunomaduro) [inline .php_cs.dist]: Please include this file and the bin folder on the analyse.

File: .php_cs.dist [hunk 3]
```diff
@@ -4,13 +4,28 @@ $finder = \PhpCsFixer\Finder::create()
...
 ;
-
+if(PHP_VERSION_ID < 50400){
 return \PhpCsFixer\Config::create()
```
Comment (nunomaduro) [inline .php_cs.dist]: Please refactor this code to avoid code duplication.

File: .php_cs.dist [hunk 4]
```diff
@@ -4,13 +4,28 @@ $finder = \PhpCsFixer\Finder::create()
...
         '@Symfony' => true,
-        'array_syntax' => ['syntax' => 'long'],
-        'increment_style' => ['style' => 'post'],
+        'array_syntax' => array('syntax' => 'long'),
+        'pre_increment' => false,
     ))
     ->setFinder($finder)
 ;
+
+}else{
```
Comment (nunomaduro) [inline .php_cs.dist]: Seems that there coding style issues.

File: .php_cs.dist (hunk 2 above)
Comment (TomKlotzPro) [reply .php_cs.dist]: Done, but perhaps check if it exist also ?

File: .php_cs.dist [hunk 5]
```diff
@@ -1,16 +1,23 @@
 <?php
```
Comment (nunomaduro) [inline .php_cs.dist]: This file should be added to git ignore.

Comment (TomKlotzPro) [reply .php_cs.dist]: done

Comment (TomKlotzPro) [reply .php_cs.dist]: but @nunomaduro if add config fil for php-cs-fixer in .gitignore how travis will be able to use it ?

**Pull request merged**